lib/livebook/intellisense/identifier_matcher.ex

```diff
@@ -207,6 +225,61 @@ defmodule Livebook.Intellisense.IdentifierMatcher do
     match_module_function(mod, hint, ctx) ++ match_module_type(mod, hint, ctx)
   end
 
+  defp match_in_struct_fields_or_local_or_var(hint, ctx) do
+    case expand_struct_fields(ctx) do
+      {:ok, struct, fields} ->
+        for {field, default} <- fields,
+            name = Atom.to_string(field),
+            ctx.matcher.(name, hint),
+            do: {:in_struct_field, struct, field, default}
+
+      _ ->
+        match_local_or_var(hint, ctx)
+    end
+  end
+
+  defp expand_struct_fields(ctx) do
+    with {:ok, quoted} <- Code.Fragment.container_cursor_to_quoted(ctx.fragment),
+         {aliases, pairs} <- find_struct_fields(quoted) do
+      mod_name = Enum.join(aliases, ".")
+      mod = expand_alias(mod_name, ctx)
+
+      fields =
+        if has_struct?(mod) do
+          map = Map.from_struct(mod.__struct__)
+
+          case ctx.type do
+            :locate ->
+              map
+
+            # When autocompleting we remove the keys that have already been filled
```

Comment:
When we are in `:locate` then `Core.Fragment.container_cursor_to_quoted` would replace the field key with `__cursor__`, and discarding other fields is fine. Or is there a case where we need this distinction?

Reply:
Ah I see! In that case I can remove the type from ctx again.

Reply:
On second thought we already have matcher in the ctx, i think type does belong there.